.github/workflows/ci.yml

```diff
@@ -0,0 +1,157 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - 'master'
+    tags:
+      - 'v*'
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+
+jobs:
+  test:
+    name: Test
+    strategy:
+      matrix:
+        # service containers are only supported on ubuntu currently
+        os:
+          - ubuntu-latest
+        py:
+          - '3.7'
+          - '3.8'
+          - '3.9'
+          # - '3.10'
+          # - '3.11.0-alpha.3'
+        db:
+          - 'mysql:5.7'
+          - 'mysql:8.0'
+          - 'mysql:latest'
+          - 'mariadb:10.2'
+          - 'mariadb:10.3'
```

Comment:
Do you really want to check against so many mariadb instances?
If yes -- it's ok but I suspect the slow execution time.
Maybe some versions can be dropped?

Reply:
currently - with all tests duplicated to include a `PYTHONASYNCIODEBUG=1` run - this has a total runtime of 6m 24s.
with the `PYTHONASYNCIODEBUG=1` runs removed i think all tests will currently be able to run in parallel, which should be able to cut this down a bit further.
to me this seems acceptable?
i've selected all currently supported database versions to ensure full coverage.

Reply:
Ok, agree

Reply:
i removed the `latest` tags though as they're redundant with the latest explicit version and won't cause unexpected test failures due to new db version releases.